lib/instance.js

```diff
@@ -642,14 +643,14 @@ Instance.prototype.save = function(options) {
 
       if (self.isNewRecord) {
         query = 'insert';
-        args = [self, self.Model.getTableName(options), values, options];
+        args = [self, self.Model.getTableName(this.$options), values, options];
       } else {
         var where = this.where();
 
         where = Utils.mapValueFieldNames(where, Object.keys(where), this.Model);
 
         query = 'update';
-        args = [self, self.Model.getTableName(options), values, where, options];
+        args = [self, self.Model.getTableName(this.$options), values, where, options];
```

Comment:
options was unused in getTableName. the code looked like this 
```javascript
Model.prototype.getTableName = function(options) { // testhint options:none
  return this.QueryGenerator.addSchema(this);
}
```
The issue here is that this.Model is attached to the prototype of the model and not the model instance itself. See https://github.com/sequelize/sequelize/blob/master/lib/model.js#L718-L719. Because of this the $schema and $schemaDelimiter attributes could be different than the values on the model that find was called on. An alternative here would be to do ``` this.Model.schema(this.$options.$schema,this.$options.$schemaDelimiter) ``` for the case where ```this.Model.$schema !== this.$options.$schema || this.Model.$schemaDelimiter !== this.$options.$schemaDelimiter ```. Does that work? If so I will revert back ALL of the options changes including making getTableName do nothing with options.